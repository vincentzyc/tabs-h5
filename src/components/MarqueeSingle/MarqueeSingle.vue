<template>
  <div class="marquee-single flex">
    <ul class="scroll-wrapper">
      <li
        v-for="(text, i) in scrollList"
        :class="['flex', 'align-middle', 'scroll-item', { anim: animate && i === 0 }]"
        :style="{ marginTop: animate && i === 0 ? '-1rem' : '' }"
      >
        <span :class="['userimg', 'flex-none', text.sex === 'man' ? 'man' : 'woman']"></span>
        <span class="scroll-text txtover">{{ text.text }}</span>
      </li>
    </ul>
  </div>
</template>

<script setup lang="ts">
let animate = $ref(false);

let animateTimeId: NodeJS.Timeout;
let scrollTimeId: NodeJS.Timeout;

const scrollList = $ref([
  {
    text: "李**抢得靓号 1305678****",
    sex: "man",
  },
  {
    text: "蔡**抢得靓号 1308288****",
    sex: "man",
  },
  {
    text: "张**抢得靓号 1326668****",
    sex: "woman",
  },
  {
    text: "罗**抢得靓号 1665688****",
    sex: "woman",
  },
  {
    text: "林**抢得靓号 1556688****",
    sex: "man",
  },
  {
    text: "周**抢得靓号 1306566****",
    sex: "woman",
  },
]);

const scroll = () => {
  if (animate) return;
  clearTimeout(animateTimeId);
  animate = true;
  animateTimeId = setTimeout(() => {
    scrollList.push(scrollList[0]);
    scrollList.shift();
    animate = false;
  }, 500);
};

onBeforeUnmount(() => {
  clearInterval(scrollTimeId);
  clearTimeout(animateTimeId);
});

onMounted(async () => {
  await nextTick();
  setTimeout(() => {
    scrollTimeId = setInterval(scroll, 2000);
  }, 10);
});
</script>
