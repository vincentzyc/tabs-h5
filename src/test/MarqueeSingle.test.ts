import { mount } from '@vue/test-utils'
import MarqueeSingle from '@/components/MarqueeSingle/MarqueeSingle.vue'

test('mount MarqueeSingle component', async () => {
  expect(MarqueeSingle).toBeTruthy()

  const wrapper = mount(MarqueeSingle)

  expect(wrapper.html()).toMatchSnapshot()
})
