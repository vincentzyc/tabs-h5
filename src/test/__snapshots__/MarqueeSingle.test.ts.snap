// Vitest Snapshot v1

exports[`mount MarqueeSingle component 1`] = `
"<div class=\\"marquee-single flex\\">
  <ul class=\\"scroll-wrapper\\">
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">李**抢得靓号 1305678****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">蔡**抢得靓号 1308288****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">张**抢得靓号 1326668****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">罗**抢得靓号 1665688****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">林**抢得靓号 1556688****</span></li>
    <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">周**抢得靓号 1306566****</span></li>
  </ul>
</div>"
`;
